### PR TITLE
Add js , jsp, jag files when creating a scan artifact for veracode scan

### LINKED ISSUE #85: Add JS, JSP and Jaggery files to veracode scan artifact.
**Description:**

Add JS, JSP and Jaggery files when creating a scanning artifact for Veracode.

### PR DESCRIPTION
**Pull request by mathuriga**
## Purpose
Add js, jsp and jaggery files in veracode scan artifact

Fixes #85

### CODE REVIEW

File: internal/automation-scripts/BuildStaticScanZip.sh [hunk 1]
```diff
@@ -48,7 +48,21 @@ do
 				find $STATIC_HOME/$product -regex $actual_pattern -exec cp {} $STATIC_HOME/$product-work \;
 			else
 				echo "$SCRIPT_TAG Copying files that match the pattern: $pattern"
-				find $STATIC_HOME/$product -name $pattern -exec cp {} $STATIC_HOME/$product-work \;
+				#Find and Copy all the js,jsp and jaggery files to scan artifact
+				if [[ $pattern == *".js"* ]] || [[ $pattern == *".jag"* ]] ;then
+            		for file in $(find $STATIC_HOME/$product -name $pattern)
+            		do
+            		    #Create same directory structure for each file
+                   		dir=$(dirname "$file")
```
Comment (ayomawdb) [inline internal/automation-scripts/BuildStaticScanZip.sh]: Shall we correct the indentations here? We need to make it same level as the comment.

File: internal/automation-scripts/BuildStaticScanZip.sh [hunk 2]
```diff
@@ -48,7 +48,21 @@ do
 				find $STATIC_HOME/$product -regex $actual_pattern -exec cp {} $STATIC_HOME/$product-work \;
 			else
 				echo "$SCRIPT_TAG Copying files that match the pattern: $pattern"
-				find $STATIC_HOME/$product -name $pattern -exec cp {} $STATIC_HOME/$product-work \;
+				#Find and Copy all the js,jsp and jaggery files to scan artifact
+				if [[ $pattern == *".js"* ]] || [[ $pattern == *".jag"* ]] ;then
+            		for file in $(find $STATIC_HOME/$product -name $pattern)
```
Comment (ayomawdb) [inline internal/automation-scripts/BuildStaticScanZip.sh]: Shall we correct the indentations here? This should be nested inside the else?

File: internal/automation-scripts/BuildStaticScanZip.sh (hunk 1 above)
Comment (mathuriga) [reply internal/automation-scripts/BuildStaticScanZip.sh]: Resolved

File: internal/automation-scripts/BuildStaticScanZip.sh (hunk 2 above)
Comment (mathuriga) [reply internal/automation-scripts/BuildStaticScanZip.sh]: Resolved

**Comment by ayomawdb** [comment]
@mathuriga seems like lot of unrelated stuff was added on top of you changes. Shall we resend a PR with only relevant changes and get it merged asap.

File: internal/automation-scripts/BuildStaticScanZip.sh [hunk 3]
```diff
@@ -48,7 +48,21 @@ do
...
 				echo "$SCRIPT_TAG Copying files that match the pattern: $pattern"
-				find $STATIC_HOME/$product -name $pattern -exec cp {} $STATIC_HOME/$product-work \;
+                        	#Find and Copy all the js,jsp and jaggery files to scan artifact
+                        	if [[ $pattern == *".js"* ]] || [[ $pattern == *".jag"* ]] ;then
+                        		for file in $(find $STATIC_HOME/$product -name $pattern)
+                        		do
+                            			#Create same directory structure for each file
+                            			dir=$(dirname "$file")
+                            			base_path=$STATIC_HOME/$product
+                            			suffix_of_dir="${dir//$base_path/}"
```
Comment (ayomawdb) [inline internal/automation-scripts/BuildStaticScanZip.sh]: Let's add comments about what is done here. Might be hard to understand later.

File: internal/automation-scripts/BuildStaticScanZip.sh [hunk 4]
```diff
@@ -48,7 +48,21 @@ do
...
+                            			target_dir=$STATIC_HOME/$product-work$suffix_of_dir
```
Comment (ayomawdb) [inline internal/automation-scripts/BuildStaticScanZip.sh]:
```suggestion
                            			target_dir=${STATIC_HOME}/${product}-work${suffix_of_dir}
```

File: internal/automation-scripts/BuildStaticScanZip.sh (hunk 3 above)
Comment (mathuriga) [reply internal/automation-scripts/BuildStaticScanZip.sh]: Resolved.

File: internal/automation-scripts/BuildStaticScanZip.sh [hunk 5]
```diff
@@ -48,7 +48,21 @@ do
 				find $STATIC_HOME/$product -regex $actual_pattern -exec cp {} $STATIC_HOME/$product-work \;
 			else
 				echo "$SCRIPT_TAG Copying files that match the pattern: $pattern"
-				find $STATIC_HOME/$product -name $pattern -exec cp {} $STATIC_HOME/$product-work \;
+                        	#Find and Copy all the js,jsp and jaggery files to scan artifact
```
[the remark this answers is not in the document]
Comment (mathuriga) [reply internal/automation-scripts/BuildStaticScanZip.sh]: Resolved

File: internal/automation-scripts/BuildStaticScanZip.sh [hunk 6]
```diff
@@ -48,7 +48,21 @@ do
...
+                        	if [[ $pattern == *".js"* ]] || [[ $pattern == *".jag"* ]] ;then
+                        		for file in $(find $STATIC_HOME/$product -name $pattern)
+                        		do
+                            			#Create same directory structure for each file
```
[the remark this answers is not in the document]
Comment (mathuriga) [reply internal/automation-scripts/BuildStaticScanZip.sh]: Resolved

**Pull request merged**